src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (num) => {
+    if (num !== undefined) {
+      sum += num;
+
+      return adder;
+    }
+
+    const result = sum;
+
+    sum = 0;
+
+    return result;
+  };
+
+  return adder;
```

Comment:
```suggestion
  const adder = (num) => {
    if (num === undefined) {
    const result = sum;

    sum = 0;

    return result;
  };

    sum += num;

    return adder;
```

If you change your check to `if (num === undefined)` and swap your operations you can get rid of the duplicate `return adder` in your code